core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -168,6 +170,40 @@ public boolean canReplicate()
         /**/;
     }
 
+    public enum JoinPushdownMode
+    {
+        OFF,
+        ON,
```

Comment:
ENABLED or EAGER or GREEDY
DISABLED or NONE

Reply:
also, add javadoc explaining that cross joins are not being pushed regardless

Reply:
I went for `DISABLED` and `ENABLED`

Reply:
changed `ENABLED` -> `EAGER` after discussion.